sphinx/domains/cpp.py

```diff
@@ -6512,10 +6512,6 @@ def add_target_and_index(self, ast: ASTDeclaration, sig: str, signode: desc_sign
             names = self.env.domaindata['cpp']['names']
             if name not in names:
                 names[name] = ast.symbol.docname
-                signode['names'].append(name)
```

Comment:
Yes, the warning came from inside docutils.

>In short: besides avoiding the warning I'm actually not sure what the specific effect of removing this line is.

In docutils world, `node['names']`  is used to create a hyperlink target that is able to refer via its name. On the other hand, Sphinx provides some roles (ex. `:cpp:func:`) to realize the cross-reference feature because docutils does not support a reference between multiple documents. Therefore, it is not needed to generate names for the hyperlink target.

Reply:
Thanks, looks good to merge.